src/sdk/actions.ts

```diff
@@ -130,3 +142,47 @@ async function returnDataToKernel(repoToken: string, stateId: string, output: ob
     },
   });
 }
+
+function createStructuredMetadata(className: string | undefined, logReturn: LogReturn) {
+  let logMessage, metadata;
+  if (logReturn) {
+    logMessage = logReturn.logMessage;
+    metadata = logReturn.metadata;
+  }
+
+  const jsonPretty = sanitizeMetadata(metadata);
+  const stackLine = new Error().stack?.split("\n")[2] ?? "";
```

Comment:
I think because you are creating a new `Error` here it's going to create a stack from within this fn. Not 100% but I was considering this in regards to my `Logs` debugging [here](https://github.com/ubiquity-os-marketplace/command-ask/pull/9#issue-2610044603).

`LogReturn` should have a `stack` from where the error happened in the plugin (if `Logs` are working properly, see the linked comment for more)